### ISSUE TITLE
Improve shaders readability

### ISSUE THREAD

**Issue by YoruZakura** [renpy/renpy#3852]
When registering shaders, the current syntax is the following (example from the documentation):
```renpy
init python:
    renpy.register_shader("example.gradient", variables="""
        uniform vec4 u_gradient_left;
        uniform vec4 u_gradient_right;
        uniform vec2 u_model_size;
        varying float v_gradient_done;
        attribute vec4 a_position;
    """, vertex_300="""
        v_gradient_done = a_position.x / u_model_size.x;
    """, fragment_300="""
        float gradient_done = v_gradient_done;
        gl_FragColor *= mix(u_gradient_left, u_gradient_right, gradient_done);
    """)
```

The fact that all the shader code goes into `str`s makes it rather hard to read, especially due to the lack of syntax coloration, and the shader being broken up (which to my understanding is for technical simplifications on the processing side).

Would it be possible that Ren'Py allows shaders to be written in separate files, so that shaders would be easier to read and write?
This could allow better interoperability with the global GLES ecosystem, and not require developers to adjust the shader code to and from Ren'Py.

I know there is no official standard for doing so, but tools like [glslang](https://www.khronos.org/opengles/sdk/tools/Reference-Compiler/) define a few common files extension that could be used:
```
Using Glslang from the Command Line

Basic use from the command line is simple. Put your shader in a file, e.g., shader.frag, and execute:

glslangValidator shader.frag

Any GLSL errors in shader.frag will be printed out. Be sure that your shader includes the appropriate #version statement; the default is #version 100, i.e., the shading language supported in OpenGL ES 2.0. The stage is indicated by the file's extension:

.vert - a vertex shader
.tesc - a tessellation control shader
.tese - a tessellation evaluation shader
.geom - a geometry shader
.frag - a fragment shader
.comp - a compute shader

There is also a non-shader file extension:

.conf - a configuration file of implementation-dependent limits

Multiple shaders can be specified on the same command line.
```

I could find online that some people also use the custom extensions `.glsl`, `.glslv`, `.glslf`, but these seem less common overall.

**Comment by renpytom** [comment]
I don't think this is likely to be something to do. Ren'Py uses a shader system in which different fragements have different priorities, and I don't think that's something that makes sense to have in existing formats with extensions. You can certainly use renpy.file to implement something like this for yourself, if you'd like, but I don't think there's a format that's so much better than the existing way that the engine should support it.

**Comment by YoruZakura** [comment]
Indeed the format itself does not bring great improvements over what is currently in Ren'Py, other than the color syntax. However due to the need to separate the `variables` and the `vertex_XX`/`fragment_XX` parameters, you can't use dumb-file strategies like the way it is done in other engines, where it pretty much does a code include and that's it.
Converting shaders that contain all the information into the three parts that Ren'Py requires can sometimes be quite challenging, so I was wondering if this process could be made easier by tweaking the engine? :) but I would understand if it was too complicated.

**Comment by hsandt** [comment]
Coming from other engines it really surprised me.

Besides syntax coloring, toggle comment currently adds a `#` instead of `//`, and it's very useful to disable shader lines and debug.

I don't need to clever shader preprocessor, a simple file with some convention to indicate the different sections would be enough for me.

Example from the shader described in OP:

```glsl
// section: variables
uniform vec4 u_gradient_left;
uniform vec4 u_gradient_right;
uniform vec2 u_model_size;
varying float v_gradient_done;
attribute vec4 a_position;

// section: vertex_300
v_gradient_done = a_position.x / u_model_size.x;

// section: fragment_300
float gradient_done = v_gradient_done;
gl_FragColor *= mix(u_gradient_left, u_gradient_right, gradient_done);
```

Example of registration API (here, I put arbitrary extension "rsl" for "Renpy shader language" but since Renpy didn't design a shader language it's not really correct, but put whatever you want, personally I'd use glsl just to enable language detection and syntax coloring in my code editor, although the file itself wouldn't be a valid glsl shader to use in other engines due to using those custom sections instead of proper functions)

```renpy
renpy.register_shader_file("example.gradient", file_path="gradient.rsl")
```

**Comment by renpytom** [comment]
@hsandt , could you file a new issue for your proposal? I like it, and don't want to lose it as a comment.
